### PR TITLE
New feature: cloudevents-poem

### PR DESCRIPTION
**Pull request by sunli829**

### CODE REVIEW

**Comment by jcrossley3** [comment]
Thanks for the contribution! I'm not really familiar with the `in_band_lifetimes` feature, but the stable toolchain obviously doesn't like it. Does poem require it?

**Comment by sunli829** [comment]
I'm very sorry, maybe the IDE has added `in_band_lifetimes` to it because of an error, and `poem` can work in stable Rust.

**Comment by jcrossley3** [comment]
> I'm very sorry, maybe the IDE has added `in_band_lifetimes` to it because of an error, and `poem` can work in stable Rust.

Awesome, thanks!

Would you mind including a small example in this PR beneath [./example-projects](https://github.com/cloudevents/sdk-rust/tree/master/example-projects). The simpler the better -- just enough to show someone how to get and post a CloudEvent using poem, similar to the apps for actix, axum, and warp in that same dir. It's neat to see the various design choices made by the different frameworks.

Please add a stanza at the bottom of [.github/workflows/rust_tests.yml](https://github.com/cloudevents/sdk-rust/blob/master/.github/workflows/rust_tests.yml) so CI can test the example, too. You can probably just cut and paste the axum one.

Thanks again! I'm looking forward to digging into poem myself at some point!

**Comment by sunli829** [comment]
Sorry I forgot to add an example. 😁

File: src/binding/poem/mod.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+//!                     "+44 2345678"
+//!                 ]
+//!             }),
+//!         )
+//!         .build()
+//!         .map_err(InternalServerError)?;
+//!     Ok(event)
+//! }
+//!
+//! let app = Route::new().at("/", post(index));
```
Comment (jcrossley3) [inline src/binding/poem/mod.rs:53]: This is cool -- can you include this in the example app, please? It looks like you copied the axum example, which I just realized doesn't return an Event for a GET. It should, just like the actix example. I should've been more specific about which example to use as a template, sorry.

File: example-projects/poem-example/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
+use cloudevents::Event;
+use poem::listener::TcpListener;
+use poem::middleware::Tracing;
+use poem::{get, handler, Endpoint, EndpointExt, Response, Route, Server};
+
+#[handler]
+async fn index_get() -> &'static str {
+    "hello from cloudevents server"
```
Comment (jcrossley3) [inline example-projects/poem-example/src/main.rs:8]: Can you make this return a `Result<Event>` instead, please? See my other comment. Thanks!

**Review by jcrossley3 — CHANGES_REQUESTED** [review]
I should've been more specific about which example to copy, sorry. The axum GET needs to return an Event like the actix example.

**Comment by sunli829** [comment]
done.😁

**Review by sunli829 — COMMENTED** [review]
done! 🙂

**Review by jcrossley3 — APPROVED** [review]

**Comment by jcrossley3** [comment]
Thank you!

**Pull request merged**